node/src/components/storage.rs

```diff
@@ -338,9 +338,7 @@ impl Storage {
                         HashingAlgorithmVersion::V1 => {
                             let _ = deleted_block_body_hashes_v1.insert(*block_header.body_hash());
                         }
-                        HashingAlgorithmVersion::V2 => {
-                            let _ = deleted_block_body_hashes_v2.insert(*block_header.body_hash());
-                        }
+                        HashingAlgorithmVersion::V2 => {}
```

Comment:
~Why is this okay for V1 blocks but not V2? Can't two V1 blocks also have identical bodies?~

I see, you're handling that below. But can't we also handle that the same way for V2 blocks?

Reply:
I removed it because it's much harder to make use of it with V2, and it's not needed for V1, either, strictly speaking.

For V1, I'm only using this list to detect bodies that have been orphaned _before_ the removal of blocks here. We're deciding which bodies to remove based on other data, anyway. Basically, if I encounter an entry in `block_body_v1_db` that is to be deleted and wasn't present in this list, it had been orphaned before.

Now, with V2, this doesn't work, because `block_body_v2_db` doesn't only hold entries for block bodies, but also pairs `[value_hash, merkle_proof]` from further down the chain. So, if while deleting entries I encounter one that isn't a hash of a body from a deleted block, it doesn't necessarily mean that it's an orphaned body - it might be a _part_ of some body that was from a deleted block. So, I can't use this list of hashes for detecting orphaned bodies, and it has no other use - which is why I removed it altogether.

Reply:
Makes sense, thanks!